## Introduction:
## The metric is safety. We are doing this project on safety to determine what neighborhood is the best. Our process was that some of our group members live in or around Pittsburgh. We were wondering what was the best neighborhood and which is the safest and how far it is compared to where we live. We did not explore any other alternatives.

In [19]:
import pandas as pd

order_data = pd.read_csv("PI.tsv", sep="\t")
order_data['INCIDENTNEIGHBORHOOD']=chip['INCIDENTNEIGHBORHOOD'].replace({'Mt. Oliver Boro':'Mt. Oliver'})
order_data['INCIDENTNEIGHBORHOOD']=chip['INCIDENTNEIGHBORHOOD'].replace({'Mount Oliver':'Mt. Oliver'})
order_data['INCIDENTNEIGHBORHOOD']=chip['INCIDENTNEIGHBORHOOD'].replace({'Mt. Oliver Neighborhood':'Mt. Oliver'})

In [20]:
print(order_data['INCIDENTNEIGHBORHOOD'].unique())
print(order_data['INCIDENTNEIGHBORHOOD'].nunique())

['North Shore' 'Upper Lawrenceville' 'Northview Heights'
 'Golden Triangle/Civic Arena' 'South Side Flats' 'Elliott'
 'Homewood North' 'South Side Slopes' 'Carrick' 'South Oakland' nan
 'Bloomfield' 'Beechview' 'Bluff' 'Strip District' 'Mount Washington'
 'East Carnegie' 'Crafton Heights' 'Marshall-Shadeland' 'East Liberty'
 'East Allegheny' 'Friendship' 'Arlington' 'Central Northside'
 'Spring Hill-City View' 'Brighton Heights' 'Westwood' 'East Hills'
 'Brookline' 'South Shore' 'Lower Lawrenceville' 'Bedford Dwellings'
 'Troy Hill-Herrs Island' 'Central Oakland' 'Garfield' 'West End'
 'West Oakland' 'Central Lawrenceville' 'Lincoln-Lemington-Belmar'
 'Sheraden' 'Beltzhoover' 'Outside City' 'Squirrel Hill South'
 'North Oakland' 'Allentown' 'Perry North' 'Homewood South' 'Shadyside'
 'Manchester' 'Squirrel Hill North' 'Regent Square' 'Homewood West'
 'Allegheny Center' 'Highland Park' 'Middle Hill' 'Overbrook' 'Hazelwood'
 'Greenfield' 'Morningside' 'Knoxville' 'Point Breeze' 'Duquesne

In [21]:
import pandas as pd

# Load the data
order_data = pd.read_csv('PI.tsv', sep='\t')
order_data['INCIDENTNEIGHBORHOOD']=chip['INCIDENTNEIGHBORHOOD'].replace({'Mt. Oliver Boro':'Mt. Oliver'})
order_data['INCIDENTNEIGHBORHOOD']=chip['INCIDENTNEIGHBORHOOD'].replace({'Mount Oliver':'Mt. Oliver'})
order_data['INCIDENTNEIGHBORHOOD']=chip['INCIDENTNEIGHBORHOOD'].replace({'Mt. Oliver Neighborhood':'Mt. Oliver'})

# Convert 'INCIDENTTRACT' to numeric, handling non-numeric values as NaN
order_data['INCIDENTTRACT'] = pd.to_numeric(order_data['INCIDENTTRACT'], errors='coerce')

# Group by 'INCIDENTNEIGHBORHOOD' and calculate the mean of 'INCIDENTTRACT' for each group
mean_incidenttract_by_neighborhood = order_data.groupby('INCIDENTNEIGHBORHOOD')['INCIDENTTRACT'].mean()

# Sort the results from lowest to highest
sorted_mean_incidenttract = mean_incidenttract_by_neighborhood.sort_values()

# Display the result
mean_incidenttract_by_neighborhood
# To ensure all rows are displayed, set pandas display option
pd.set_option('display.max_rows', None)

# Print the result
print(sorted_mean_incidenttract)


INCIDENTNEIGHBORHOOD
Bluff                           103.000000
Central Business District       201.000000
Strip District                  203.000000
Crawford-Roberts                334.823922
South Oakland                   409.000000
West Oakland                    467.457795
Terrace Village                 510.138870
Central Oakland                 549.130235
Bedford Dwellings               550.575701
Golden Triangle/Civic Arena     562.906265
Middle Hill                     563.428364
Upper Hill                      576.369807
North Oakland                   580.901512
Polish Hill                     650.202925
Lower Lawrenceville             693.021505
Shadyside                       749.386527
Friendship                      895.233333
Bloomfield                      906.880974
Central Lawrenceville           972.522235
Upper Lawrenceville            1011.000000
Morningside                    1014.000000
Stanton Heights                1044.553846
Garfield                       10

In [5]:
sorted_mean_incidenttract.head(10)

INCIDENTNEIGHBORHOOD
Bluff                          103.000000
Central Business District      201.000000
Strip District                 203.000000
Crawford-Roberts               334.823922
South Oakland                  409.000000
West Oakland                   467.457795
Terrace Village                510.138870
Central Oakland                549.130235
Bedford Dwellings              550.575701
Golden Triangle/Civic Arena    562.906265
Name: INCIDENTTRACT, dtype: float64

## Individual Conclusiomn
## The code presented for this data set loads the data from the PI.tsv, which stands for Police Incidents. Then, Convert 'INCIDENTTRACT' to numeric, handling non-numeric values. The next lines of code group by 'INCIDENTNEIGHBORHOOD' and calculate the mean of 'INCIDENTTRACT' for each group. I also checked and fixed overlaps. Lastly, I displayed and printed the results and reprinted the top ten safest neighborhoods with the least incidents. Based off the mean from each neighborhood, my anaylsis is that the safest neighborhood is Bluff with an average 103 police incidents. In second, the Central Business District with an average of 201 police incidents. The third safest neighborhood based off police incidents is Strip District with an average of 203. The rest of the top ten are displayed above.